dask/dataframe/io/parquet.py

```diff
@@ -229,8 +239,33 @@ def to_parquet(path, df, compression=None, write_index=None, has_nulls=None,
                                            fixed_text=fixed_text,
                                            object_encoding=object_encoding)
 
+    if append:
+        pf = fastparquet.api.ParquetFile(path, open_with=myopen)
+        if pf.file_scheme != 'hive':
+            raise ValueError('Requested file scheme is hive, '
+                             'but existing file scheme is not.')
+
+        assert (all(pf.columns == df._meta.columns),
+                'Appended columns not exactly the same.\n'
+                'New: {} | Previous: {}'.
+                format(pf.columns,list(df._meta.columns)))
```

Comment:
If we have `write_index == True`, then we do `df = df.reset_index()` and `df._meta.columns` contains the index as well. If we don't write the index then the parquet file shouldn't contain one as well. I assumed we are appending dfs with the same properties.